sdk/src/main/scala/coop/rchain/sdk/dag/merging/DagMergingLogic.scala

```diff
@@ -104,9 +104,10 @@ object DagMergingLogic {
       conflicts: (D, D) => Boolean,
       depends: (D, D) => Boolean
   ): (Map[D, Set[D]], Map[D, Set[D]]) = {
-    val conflictsMap = computeConflictsMap(conflictSet, finalSet, conflicts) ++
+    val conflictsMap = computeConflictsMap(conflictSet, finalSet, conflicts) |+|
       computeConflictsMap(conflictSet, conflictSet, conflicts)
-    val dependencyMap = computeDependencyMap(conflictSet, finalSet, depends)
+    val dependencyMap = computeDependencyMap(conflictSet, finalSet, depends) |+|
+      computeDependencyMap(conflictSet, conflictSet, depends)
```

Comment:
Here no dependencies inside conflict were not accounted for calculation of the merge set. Not sure why I was thinking this way (I even explicitly specified this in a test). Now dependencies required are dependencies inside conflict set, plus between conflict set and final set. Test is adjusted.